src/V3Graph.h

```diff
@@ -172,7 +172,106 @@ class V3Graph VL_NOT_FINAL {
 
 //============================================================================
 
+// Holds list of types. For use mainly in compile-time code generation.
+template <typename... TN>
+struct VTypeList {
+    template <typename... UN>
+    constexpr VTypeList<TN..., UN...> operator+(VTypeList<UN...>) const {
+        return {};
+    }
+};
+
+namespace internal_vl {
+
+// Holds one type. Can be safely used as a return or argument type without any potential
+// limitations of held type.
+template <typename T>
+struct VTypeWrapper {
+    using type = T;
+};
+
+template <typename T0, typename... TN>
+static inline constexpr VTypeWrapper<T0> vlTlFront(VTypeList<T0, TN...>) {
+    return {};
+}
+
+}  // namespace internal_vl
+
+// Alias for front type held by type list `TL`.
+template <typename TL>
+using VTypeListFront = typename decltype(internal_vl::vlTlFront(TL{}))::type;
+
+template <typename TL1, typename TL2>
+using VJoinedTypeLists = decltype(TL1{} + TL2{});
+
+namespace internal_vl {
+
+inline static constexpr bool isClassIdOfOneOf(uintptr_t id, VTypeList<>) VL_PURE { return false; }
+
+template <typename Base0, typename... BaseN>
+inline static constexpr bool isClassIdOfOneOf(uintptr_t id, VTypeList<Base0, BaseN...>) VL_PURE {
+    return id == Base0::rttiClassId() || isClassIdOfOneOf(id, VTypeList<BaseN...>{});
+}
+
+#define INTERNAL_VL_RTTI_COMMON_IMPLEMENTATION(ThisClass) \
```

Comment:
Maybe put _INTERNAL at end?

Reply:
For consistency with namespace and header name (`V3RttiInternal`/`V3Rtti.h`), I've changed it to `V3RTTIINTERNAL_VL_RTTI_COMMON_IMPL`.